Modules/unicodedata.c

```diff
@@ -90,22 +102,25 @@ static PyMemberDef DB_members[] = {
         {NULL}
 };
 
-/* forward declaration */
-static PyTypeObject UCD_Type;
-#define UCD_Check(o) Py_IS_TYPE(o, &UCD_Type)
+static inline int UCD_Check(PyObject *o) {
+    return PyModule_Check(o) ? 0 : 1;
```

Comment:
@vstinner this one is not obvious.  The clients of UCD_Check are the methods listed in `unicodedata_functions`

These functions can be called in three ways:

1. As methods on the module
2. As methods on the ucd type
3. As methods exported using a capsule (codecs.c uses it).

All calls to UCD_Check in these methods are short circuited with a null check for `o`.  It turns out that all of the calls from use case **3** come with o=NULL.  So essentially the contract of UCD_Check is to return 1 if it is called from **2** otherwise to return 0 if called from **1**. Prior to this change, with the ucd type being a static C variable, the cheapest check was Py_IS_TYPE(o, &UCD_Type).   Now, it is cheaper to check the opposite case (and return the opposite int) with PyModule_Check